### ISSUE TITLE
Support of Terraform 1.x - Quote with pushvars

### ISSUE THREAD

**Issue by snahelou** [hashicorp-community/tf-helper#36]
Hello

First of all, thanks for your awesome helper. You can't imagine the time save for us :) 

I think that the way of you fetch values using var-files seems not compatible with TF1.x: 

Example vars-file: 
```
environment = "prod"
```

```
$ tfswitch
Switched terraform to version "0.13.7"

$ tfh pushvars -name xxxx -var-file /tmp/vars -overwrite-all -dry-run
Processing environment type:terraform hcl:false sensitive:false value:prod

$ tfswitch
Switched terraform to version "1.0.1"

$ tfh pushvars -name xxxx -var-file /tmp/vars -overwrite-all -dry-run
Processing environment type:terraform hcl:false sensitive:false value:\"prod\"
```

I don't know how to fix this but this issue is related to `terraform console` imho

BR

**Comment by fprimex** [comment]
Hi, sorry for the delay on responding. This is a formatting change going from 0.13 to 0.14, and there is also an issue with the ordering of arguments going to `terraform console`.

```
$ terraform version
Terraform v0.13.7

$ echo 'var.a' | terraform console
foo
```

```
$ terraform version
Terraform v1.0.5
on darwin_amd64

$ echo 'var.a' | terraform console
"foo"
```

This problem is specific to loading variables using the configuration option, such as `tfh pushvars .`. There is another issue with `pushvars` in that `terraform` no longer accepts a configuration directory as a target. Instead, all operations are performed in the current working directory, but if you want Terraform to change directory first you give it `-chdir=<path>`.

Two things will need to happen to resolve the issue. One, string detection will need to be done before escaping values so quotes can be stripped, although there is a case where a value could have literal quotes and be marked as HCL, in which case they should be kept.

Secondly, the `terraform console` command will need to be updated with `-chdir` in the format: `echo "var.a" | terraform -chdir=<path> console`, or otherwise changed.

My preference here would be to revamp the next version to require 1.x+, and dispense with needing to do a bunch of checks for earlier versions.

I haven't had a ton of time to work on `tf-helper`, however I'm hoping that changes in the very near future with some different job responsibilities I am taking on.

All that said, if you need this to work *today* then I believe this will do it:

```
    for var in $all_vars; do
      if   [ -n "$var_file_arg" ] && [ -n "$config_dir" ]; then
        val_lines="$(echo "var.$var" \
                     | terraform console -chdir="$config_dir" "$var_file_arg" 2>&1)"
      elif [ -n "$var_file_arg" ] && [ -z "$config_dir" ]; then
        val_lines="$(echo "var.$var" \
                     | terraform console "$var_file_arg" 2>&1)"
      elif [ -z "$var_file_arg" ] && [ -n "$config_dir" ]; then
        val_lines="$(echo "var.$var" \
                     | terraform console -chdir="$config_dir" 2>&1)"
      else
        val_lines="$(echo "var.$var" \
                     | terraform console 2>&1)"
      fi

      if [ 0 -ne $? ]; then
        echodebug "Unable to retrieve value for potential variable $var"
        echodebug "Stdout of terraform console was:"
        echodebug "$val_lines"
        continue
      fi

      # Remove texts related to state lock from the output if present.
      val_lines="$(echo "$val_lines" \
                   | grep -vE "Acquiring state lock|Releasing state lock" 2>&3)"

      val_lines="$(echo "$val_lines" | sed -e 's/^"//; s/"$//' 2>&3)"
```

Everywhere `$config_dir` is used it changes to `-chdir="$config_dir"`, and then beginning and ending quotes are stripped, which should not affect maps or lists since they start with `{` and `[`.

**Comment by kvendingoldo** [comment]
btw. you can also use [tenv](https://github.com/tofuutils/tenv) that support Terraform as well as OpenTofu (and Terragrunt :) ) in one tool. It allow you to simplify version management and can do much more, than tfswitch.

**Comment by yermulnik** [comment]
Spammer ⬆️ 
* https://github.com/warrensbox/terraform-switcher/issues/175#issuecomment-2034701614
* https://github.com/warrensbox/terraform-switcher/issues/315#issuecomment-2033189467

**Comment by brenthc** [comment]
ty @yermulnik
